### PR TITLE
New Features

### PR DESCRIPTION
**Pull request by ecwillis**

### CODE REVIEW

File: conf/default.yaml [hunk 1]
```diff
@@ -81,3 +81,28 @@ deploy:
       - '5000'
     volume:
       - '/var/run/docker.sock:/var/run/docker.sock'
+
+mongo:
+  enabled: true
```
Comment (jgallen23) [inline conf/default.yaml]: lets disable by default

File: conf/default.yaml [hunk 2]
```diff
@@ -81,3 +81,28 @@ deploy:
...
+  image: 'mongo'
+  args:
+    name: 'mongo'
+    d: true
+    restart: 'always'
+    volume:
+      - '/data/db'
+
+letsencrypt:
+  enabled: true
```
Comment (jgallen23) [inline conf/default.yaml]: disable by default

File: example/server.yaml [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
   instanceType: 't2.nano'
   instanceSize: 8
-  keyName: 'default'
+  keyName: 'default-test'
   users:
-    - 'jgallen23'
+    - 'ecwillis'
+
+ec2:
+  SubnetId: 'subnet-4fefa02a'
```
Comment (jgallen23) [inline example/server.yaml]: why did you need this?

File: lib/create.js [hunk 4]
```diff
@@ -19,7 +19,9 @@ module.exports = function(config, allDone) {
     ec2: (done) => {
       ec2.run(done);
     },
-
+    dns: (done) => {
```
Comment (jgallen23) [inline lib/create.js]: eh?

File: lib/ec2.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ class EC2 {
         log(['debug'], 'Checking if keypair exists');
         const params = {
           KeyNames: [
-            this.config.server.keypair,
+            this.config.server.keyName,
```
Comment (jgallen23) [inline lib/ec2.js]: keypair is valid, why did this change?

File: lib/ec2.js [hunk 6]
```diff
@@ -33,7 +33,7 @@ class EC2 {
       },
 
       keypair: ['hasKeypair', (results, done) => {
-        const name = this.config.server.keypair;
+        const name = this.config.server.keyName;
```
Comment (jgallen23) [inline lib/ec2.js]: same as above

File: example/server.yaml (hunk 3 above)
Comment (ecwillis) [reply example/server.yaml]: This was for my account which is old and doesn't have default subnet ids for the vpc. There is documentation out there somewhere, but if you don't include this - the command fails.

Comment (jgallen23) [reply example/server.yaml]: lets remove from the example

File: lib/ec2.js [hunk 7]
```diff
@@ -168,6 +171,10 @@ touch /tmp/provision_complete`;
         done(null, instance.PublicDnsName);
       }],
 
+      dns: ['host', (results, done) => {
+        this.route53.run(results.host, 'create', done);
```
Comment (jgallen23) [inline lib/ec2.js]: we should add a way to disable this

File: example/server.yaml [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
 env: 'production'
 server:
-  name: 'test'
+  name: 'stage-apps'
   instanceType: 't2.nano'
   instanceSize: 8
   keypair: 'default'
   users:
-    - 'jgallen23'
+    - 'jgallen'
```
Comment (jgallen23) [inline example/server.yaml:9]: not the right user

File: lib/route53.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+class Route53 {
+  constructor(config) {
+    this.config = config;
+    this.route53 = new AWS.Route53({
+      region: this.config.server.region
+    });
+  }
+
+  run(ipAddress, action, allDone) {
+    async.auto({
```
Comment (jgallen23) [inline lib/route53.js:20]: should we check if its already set first?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**